### ISSUE TITLE
Site Cloner not generating index.html and unable to update setoolkit on Kali

### ISSUE THREAD

**Issue by vedtam** [trustedsec/social-engineer-toolkit#206]
Hi,

I have just reinstalled the latest Kali Linux and while trying to launch setoolkit I get this first:

```
[*] Kali bleeding edge was not detected to be on...
[*] Kali install detected. Note that if you are not using bleeding edge repositories, your version of SET will be roughly 4 months behind.
[*] It is recommended to switch to bleeding-edge repos to ensure you are running the latest version of SET and other tools.
Press [enter] to accept that SET is several months out of date and probably contains bugs and issues.
```

I have added the repo by:

```
echo deb http://http.kali.org/kali kali-bleeding-edge contrib non-free main >> /etc/apt/sources.list
apt-get update
apt-get upgrade
```

But I still got the error message when starting setoolkit. Anyway I hit enter and try to run  3) Credential Harvester Attack Method. After I fish the wizard in my /var/www/ I got only the post.php file, no harvest.txt no index.html

As a last hope I have tried to remove and eventually reinstal, but here I get another error:

```
root@kali:~# apt-get remove setoolkit
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package setoolkit
```

What could be possibly wrong? Kali version is (x64):
```
root@kali:~# uname -a
Linux kali 4.3.0-kali1-amd64 #1 SMP Debian 4.3.5-1kali1 (2016-02-11) x86_64 GNU/Linux
```

**Comment by L1ghtn1ng** [comment]
As you are on the kali rolling you did not need to enable the bleeding edge repo. You just need to update your system. Remove that repo as it will cause issues

**Comment by trustedsec** [comment]
Looks like an old version too - site cloner was fixed a long time ago. Move to rolling.

**Comment by vedtam** [comment]
I have removed bleeding edge repo then apt-get update/upgrade. Error message still prezent at startup.

this is how my sources.list looks:
```

deb http://http.kali.org/kali kali-rolling main contrib non-free

deb http://http.kali.org/kali sana main non-free contrib
deb http://security.kali.org/kali-security sana/updates main contrib non-free
```

**Comment by trustedsec** [comment]
What version of SET does it say you are at?

**Comment by L1ghtn1ng** [comment]
You need to remove all sana repos and add just kali rolling, please read http://docs.kali.org/general-use/kali-linux-sources-list-repositories

**Comment by L1ghtn1ng** [comment]
Also you need to a dist-upgrade not the plain upgrade

**Comment by vedtam** [comment]
Still no luck :(

sources.list

`deb http://http.kali.org/kali kali-rolling main contrib non-free`

1.) apt-get update
2.) apt-get dist-upgrade

There is the error still at startup and won't generate index.html or harvest.txt. Version of setoolkit is Version: 7.0.2

```
root@kali:~# setoolkit
[*] Kali bleeding edge was not detected to be on...
[*] Kali install detected. Note that if you are not using bleeding edge repositories, your version of SET will be roughly 4 months behind.
[*] It is recommended to switch to bleeding-edge repos to ensure you are running the latest version of SET and other tools.
Press [enter] to accept that SET is several months out of date and probably contains bugs and issues.
```

**Comment by trustedsec** [comment]
Yeah - not really sure what you are running into. Credential harvester works great on the latest Kali and on 7.0.2..

If you want, you can post the full log of options that you use and I can try to re-create..

**Comment by L1ghtn1ng** [comment]
think it might be an idea of downloading the rolling ISO from kali.org and doing a re-install and see if you hit it after that
On 15 Feb 2016 4:46 p.m., trustedsec <notifications@github.com> wrote:Yeah - not really sure what you are running into. Credential harvester works great on the latest Kali and on 7.0.2..

If you want, you can post the full log of options that you use and I can try to re-create..

**Comment by vedtam** [comment]
I will try to pull a full log. I would be grateful if you could suggest a method for exporting a log.

**Comment by L1ghtn1ng** [comment]
Post it on pastebin and then in the github message post the link

**Comment by vedtam** [comment]
Hi,

I did a clean install of Kali using the latest ISO. After all the apt-gets and dist-upgrades I still receive the mentioned error at startup.

Could you be so kind to tell me how I could pull a log on the app? I was looking everywhere but I can't figure it out.

Thanks

**Comment by Untamedwolf** [comment]
hi! I'm not able change ip while performing harvester attack. It does not allow me to use external IP instead uses the internal IP by default   
Any help?

**Comment by MASK7LJ** [comment]
i am getting this text when i am trying to clone any website:

[*] This could take a little bit...

The best way to use this attack is if username and password form
fields are available. Regardless, this captures all POSTs on a website.
[*] The Social-Engineer Toolkit Credential Harvester Attack
[*] Credential Harvester is running on port 80
[*] Information will be displayed to you as it arrives below:
---so what should do to fix it??

**Comment by vraghuvaran** [comment]
I'm using the setoolkit for the first time .Where will all the credentials list file that the user  has entered be located in kali.plz help me out.

**Comment by Dabisco** [comment]
I also have that problem, any help?

**Comment by rutera** [comment]
The problem is still exists. 
My version is:
Version: 7.7.9 - last kali version 2018.4 and upgrade full

1>2>3>2
[*] You may need to copy /var/www/* into /var/www/html depending on where your directory structure is.
Press {return} if you understand what we're saying here.
And not a clone site, check permission per folder is ok.

I find partially solutions, clone site is here:
/root/.set/web_clone
but is still processing editing for a working ...

any help?

**Comment by chenlonghui1234** [comment]
> The problem is still exists.
> My version is:
> Version: 7.7.9 - last kali version 2018.4 and upgrade full
> 
> 1>2>3>2
> [_] You may need to copy /var/www/_ into /var/www/html depending on where your directory structure is.
> Press {return} if you understand what we're saying here.
> And not a clone site, check permission per folder is ok.
> 
> I find partially solutions, clone site is here:
> /root/.set/web_clone
> but is still processing editing for a working ...
> 
> any help?

I have the same problem. How can I solve this problem?

**Comment by krn420** [comment]
Hey bro chenlonghui1234 ,
i have same as you i.e
this problem causes due to your setoolkit uses the python inbuilt server instead of apache.

SOLUTION: simple turn ON apache service from configuration.
open terminal and enter command 
leafpad /etc/setoolkit/set.config
                                             <search for APACHE_SERVER in config file>
set 
APACHE_SERVER=OFF to ON i.e APACHE_SERVER=ON
save its file and reuse the whole process of cloning.

if problem gets solve reply me man.

**Comment by hueli54** [comment]
> NOTE :
> this bad line is still there after the upper solution . but SETOOLKIT works
> [] You may need to copy /var/www/ into /var/www/html depending on where your directory structure is.
> Press {return} if you understand what we're saying here.
> simply press ENTER
> 
> after that apache server set to ON and enjoy the hack

I get the same thing and I give it in full, it saves the email but the password does not. Password = false Or also Password = true But don't show it to me.

**Comment by hueli54** [comment]
> NOTE :
> this bad line is still there after the upper solution . but SETOOLKIT works
> [] You may need to copy /var/www/ into /var/www/html depending on where your directory structure is.
> Press {return} if you understand what we're saying here.
> simply press ENTER
> 
> after that apache server set to ON and enjoy the hack

I get the same thing and I give it enter, it saves the email but the password does not. It only appears like this: Password = false Or also Password = true But don't show it to me.

**Comment by hueli54** [comment]
> > NOTE :
> > this bad line is still there after the upper solution . but SETOOLKIT works
> > [] You may need to copy /var/www/ into /var/www/html depending on where your directory structure is.
> > Press {return} if you understand what we're saying here.
> > simply press ENTER
> > after that apache server set to ON and enjoy the hack
> 
> I get the same thing and I give it enter, it saves the email but the password does not. It only appears like this: Password = false Or also Password = true But don't show it to me.

Array
(
    [jazoest] => 2675
    [lsd] => AVr_9AkV
    [display] => 
    [enable_profile_selector] => 
    [isprivate] => 
    [legacy_return] => 0
    [profile_selector_ids] => 
    [return_session] => 
    [skip_api_login] => 
    [signed_next] => 
    [trynum] => 1
    [timezone] => 285
    [lgndim] => eyJ3Ijo4MDAsImgiOjYwMCwiYXciOjgwMCwiYWgiOjU2OSwiYyI6MjR9
    [lgnrnd] => 095601_GiyE
    [lgnjs] => 1588266484
    [email] => palacios@gmail.com
    [prefill_contact_point] => palaci
    [prefill_source] => browser_onload
    [prefill_type] => contact_point
    [first_prefill_source] => browser_onload
    [first_prefill_type] => contact_point
    [had_cp_prefilled] => true
    [had_password_prefilled] => false
    [ab_test_data] => AAAAAAVAAV/VAVVVAAAAVAVAVAAAVVAAAAAAAVAAAX/XvHAAAAICAB
    [ep] => #PWD_BROWSER:5:1588266502:AXpQAKmVtp0BDjJHOvjO3UHxRHZkpqhGdklsNLpiq7CWmi1Tod94NYTnmfJInPuV1PXNTYLTtvT621H+xzuqGG1zTHG+dfTPFZL0+W7PEgEgtNIKDJEIy6VmBOxPu5E35YNniR7MLK+19Zfs
)

how can i solve that
